spec/models/info_request_batch_spec.rb

```diff
@@ -101,23 +101,21 @@
       end
     end
 
-    it 'should send requests to requestable public bodies, and return a list of unrequestable ones' do
-      allow(first_public_body).to receive(:is_requestable?).and_return(false)
-      unrequestable = info_request_batch.create_batch!
-      expect(unrequestable).to eq([first_public_body])
+    it 'should send requests to requestable public bodies' do
+      expect(info_request_batch).to receive(:requestable_public_bodies).
+        and_return([first_public_body])
+      info_request_batch.create_batch!
       expect(info_request_batch.info_requests.size).to eq(1)
       request = info_request_batch.info_requests.first
       expect(request.outgoing_messages.first.status).to eq('sent')
     end
 
     it 'should not only send requests to public bodies if already sent' do
-      allow(second_public_body).to receive(:is_requestable?).and_return(false)
       request = info_request_batch.info_requests = [
         FactoryGirl.create(:info_request, public_body: first_public_body)
       ]
-      unrequestable = info_request_batch.create_batch!
-      expect(unrequestable).to eq([second_public_body])
-      expect(info_request_batch.info_requests.size).to eq(1)
+      info_request_batch.create_batch!
+      expect(info_request_batch.info_requests.size).to eq(2)
```

Comment:
The expected value here has changed from `1` to `2`. Shouldn't it remain as `1` in this case?

To me, it looks like we start off with a Batch with a request sent to a PublicBody. We then call `create_batch!` again, presumably to check that the PublicBody doesn't receive the request again.

This spec implies that it _should_ send twice?

Reply:
Ah, so what's happening here is that we're now saying is:

Given the batch already has 1 request created,
When I send the batch
Then I'll have a _total_ of 2 requests sent (rather than 3, if we'd resent to the body of the existing request).

Reply:
I think this is kinda confusing.

I'd rewrite this spec entirely as:

```ruby
it 'does not resend requests to public bodies that have already received the request' do
  info_request_batch = FactoryBot.create(:info_request_batch, :sent)
  info_request_batch.create_batch!
  expect(info_request_batch.info_requests.count).to eq(1)
end
```

Requires https://github.com/mysociety/alaveteli/pull/4712

Reply:
To make it even clearer I'm now checking `info_request_batch.info_requests.count` increases by 1. Changed in 590e1e9826eb6c365903ef6f1c3fa91f9809f444